src/Microsoft.ML.Core/Data/IHostEnvironment.cs

```diff
@@ -129,6 +135,11 @@ public interface IHost : IHostEnvironment
     /// generators are NOT thread safe.
     /// </summary>
     Random Rand { get; }
+
+    /// <summary>
+    /// The random source backing <see cref="Rand"/>.
+    /// </summary>
+    IRandomSource RandomSource { get; }
```

Comment:
This is a breaking change since IHost is a public interface.  Does this need to be a new property, or can it be implemented as a derived `Random` type and just pass that in to the existing APIs that accept a `Random`?

Reply:
@asp2286 can you please comment?  Can you derive from `Random`?